### PR TITLE
refactor(dli/package): the package resource reconstruction

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

The API [document](https://support.huaweicloud.com/intl/en-us/api-dli/dli_02_0130.html ) supports no grouping and **modelFile** type features, so refactoring the package resource.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
1. the `group_name` parameter is changed is from optional to required.
2. the `type` parameter supports new type `modelFile`.
3. delete other upload interfaces and use [Uploading a Package Group](https://support.huaweicloud.com/intl/en-us/api-dli/dli_02_0130.html) interface  to upload group resource.
4. If `group_name` is specified, the ID is constructed from the `group_name` and `object_name`,
  the format is `<group_name>#<object_name>`, otherwise the resource ID which equals the `object_name`.

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliPackage_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliPackage_ -timeout 360m -parallel 4
=== RUN   TestAccDliPackage_basic
=== PAUSE TestAccDliPackage_basic
=== RUN   TestAccDliPackage_not_groupName
=== PAUSE TestAccDliPackage_not_groupName
=== RUN   TestAccDliPackage_owner
=== PAUSE TestAccDliPackage_owner
=== CONT  TestAccDliPackage_basic
=== CONT  TestAccDliPackage_owner
=== CONT  TestAccDliPackage_not_groupName
--- PASS: TestAccDliPackage_not_groupName (47.71s)
--- PASS: TestAccDliPackage_basic (48.07s)
--- PASS: TestAccDliPackage_owner (48.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       48.790s
```

### CODE REVIEW

File: docs/resources/dli_package.md [hunk 1]
```diff
@@ -29,13 +29,14 @@ The following arguments are supported:
...
 
-* `group_name` - (Required, String, ForceNew) Specifies the group name which the package belongs to.
+* `group_name` - (Optional, String, ForceNew) Specifies the group name which the package belongs to.
   Changing this parameter will delete the current package and upload a new package.
 
 * `type` - (Required, String, ForceNew) Specifies the package type.
   + **jar**: `.jar` or jar related files.
   + **pyFile**: `.py` or python related files.
   + **file**: Other user files.
+  + **modelFile**: User AI model file.
```
Comment (Lance52259) [inline docs/resources/dli_package.md]:
```suggestion
  + **modelFile**: User AI model files.
```

Comment (wuzhuanhong) [reply docs/resources/dli_package.md]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_package.go [hunk 2]
```diff
@@ -114,74 +92,60 @@ func buildDliDependentPackageCreateOpts(d *schema.ResourceData) resources.Create
...
 
 	return ResourceDliDependentPackageV2Read(ctx, d, meta)
 }
 
-func getGroupNameAndPackageName(id string) (groupName, packageName string, err error) {
-	names := strings.Split(id, "/")
+func getGroupNameAndPackageName(id, group string) (groupName, packageName string, err error) {
+	if group == "" {
+		return "", id, nil
+	}
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_package.go]: using split() function and if the length of result is equals one, just set the package name and return. If the the length is greater than one, set the element which the index is zero to the group name and set the element which the index is one to the package name.

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliPackage_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliPackage_ -timeout 360m -parallel 4
=== RUN   TestAccDliPackage_basic
=== PAUSE TestAccDliPackage_basic
=== RUN   TestAccDliPackage_not_groupName
=== PAUSE TestAccDliPackage_not_groupName
=== RUN   TestAccDliPackage_owner
=== PAUSE TestAccDliPackage_owner
=== CONT  TestAccDliPackage_basic
=== CONT  TestAccDliPackage_owner
=== CONT  TestAccDliPackage_not_groupName
--- PASS: TestAccDliPackage_not_groupName (46.84s)
--- PASS: TestAccDliPackage_owner (47.46s)
--- PASS: TestAccDliPackage_basic (48.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       48.925s

File: huaweicloud/services/dli/resource_huaweicloud_dli_package.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_package.go]: fixed

**Pull request merged**